e2etest/newe2e_resource_managers_blobfs.go

```diff
@@ -0,0 +1,369 @@
+package e2etest
+
+import (
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/blob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azdatalake/directory"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azdatalake/file"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azdatalake/filesystem"
+	datalakeSAS "github.com/Azure/azure-sdk-for-go/sdk/storage/azdatalake/sas"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azdatalake/service"
+	"github.com/Azure/azure-storage-azcopy/v10/cmd"
+	"github.com/Azure/azure-storage-azcopy/v10/common"
+	"runtime"
+)
+
+// check that everything aligns with interfaces
+func init() {
+	void := func(_ ...any) {} // prevent go from erroring from unused vars
+
+	var sm ServiceResourceManager = &BlobFSServiceResourceManager{}
+	var cm ContainerResourceManager = &BlobFSFileSystemResourceManager{}
+	var om ObjectResourceManager = &BlobFSPathResourceProvider{}
+
+	var rrm RemoteResourceManager
+
+	rrm = &BlobFSServiceResourceManager{}
+	rrm = &BlobFSFileSystemResourceManager{}
+	rrm = &BlobFSPathResourceProvider{}
+
+	void(rrm, sm, cm, om)
+}
+
+func dfsStripSAS(uri string) string {
+	parts, err := datalakeSAS.ParseURL(uri)
+	common.PanicIfErr(err)
+
+	parts.SAS = datalakeSAS.QueryParameters{} // remove SAS
+
+	return parts.String()
+}
+
+type BlobFSServiceResourceManager struct {
+	Account AccountResourceManager
+
+	internalClient *service.Client
+}
+
+func (b *BlobFSServiceResourceManager) Location() common.Location {
+	return common.ELocation.BlobFS()
+}
+
+func (b *BlobFSServiceResourceManager) Level() cmd.LocationLevel {
+	return cmd.ELocationLevel.Service()
+}
+
+func (b *BlobFSServiceResourceManager) URI() string {
+	return dfsStripSAS(b.internalClient.DFSURL())
+}
+
+func (b *BlobFSServiceResourceManager) ValidAuthTypes() ExplicitCredentialTypes {
+	return EExplicitCredentialType.With(EExplicitCredentialType.OAuth(), EExplicitCredentialType.SASToken(), EExplicitCredentialType.AcctKey())
```

Comment:
why do we support account key in dfs? Is it something we should consider deprecating?

Reply:
We probably should deprecate it in the future yes.

It's supported because back in the day SAS tokens weren't supported